.travis.yml

```diff
@@ -55,3 +58,18 @@ after_failure:
   - cat test4/logs/all-logs.log
   - cat test5/logs/all-logs.log
   - cat test6/logs/all-logs.log
+
+before_deploy:
+  - docker login -u "$DOCKER_USER" -p "$DOCKER_PASSWORD"
+  - IMAGE_NAME="$TRAVIS_REPO_SLUG"
+  - TAG=`echo $TRAVIS_BRANCH | sed 's,^master$,latest,'`
+  - docker build -t "${IMAGE_NAME}:alpine-$TAG" -f support/docker/production/Dockerfile.alpine
+  - docker build -t "${IMAGE_NAME}:stretch-$TAG" -f support/docker/production/Dockerfile.stretch
+  - docker tag "${IMAGE_NAME}:stretch-$TAG" "${IMAGE_NAME}:$TAG"
+
+deploy:
+  provider: script
+  script: docker push "${IMAGE_NAME}:$TAG" && docker push "${IMAGE_NAME}:alpine-$TAG" && docker push "${IMAGE_NAME}:stretch-$TAG"
+  on:
+    all_branches: true
```

Comment:
Shouldn't it be only on master in the end? (or is it only there while you are testing?)